lexers/hare.lua

```diff
@@ -14,47 +14,57 @@ lex:add_rule('whitespace', token(lexer.WHITESPACE, lexer.space^1))
 
 -- Keywords.
 lex:add_rule('keyword', token(lexer.KEYWORD, word_match{
-  'as', 'break', 'case', 'const', 'continue', 'def', 'defer', 'else', 'export', 'false', 'fn',
-  'for', 'if', 'is', 'let', 'match', 'null', 'nullable', 'return', 'static', 'struct', 'switch',
-  'true', 'type', 'use', 'yield'
+  'as', 'break', 'case', 'const', 'continue', 'def', 'defer', 'else', 'export', 'false', 'fn', 'for',
+  'if', 'is', 'let', 'match', 'null', 'nullable', 'return', 'static', 'struct', 'switch', 'true',
+  'type', 'use', 'yield'
 }))
 
 -- Functions.
+local size_builtin = 'size' * lexer.space^0 * #P('(')
 lex:add_rule('function', token(lexer.FUNCTION, word_match{
-  'len', 'alloc', 'free', 'assert', 'abort', 'size', 'append', 'insert', 'delete', 'vastart',
-  'vaarg', 'vaend'
-}))
+  'abort', 'align', 'alloc', 'append', 'assert', 'delete', 'free', 'insert', 'len', 'offset',
+  'vaarg', 'vaend', 'vastart'
+} + size_builtin))
```

Comment:
Unless Hare has first-class functions, may I suggest including `'size'` in the list and tack on `#(lexer.space^0 * '(')` after `word_match`? Otherwise, you should have `#(lexer.space^0 * '(')` so you don't include whitespace in the match.

Reply:
> Unless Hare has first-class functions, may I suggest including `'size'` in the list and tack on `#(lexer.space^0 * '(')` after `word_match`?

Requiring '(' after builtins will result in less than ideal highlighting of cases like

```
let len = 42;
```

Here `len` would be highlighted as a regular identifier because it's not followed by parenthesis, but it's actually a hard keyword and can't be used in this context.

> Otherwise, you should have `#(lexer.space^0 * '(')` so you don't include whitespace in the match.

Thanks for the suggestion, I'll add this